projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -4856,14 +4856,14 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
         if (args.context['templateID'] === 'dataRow' && args.context['$implicit'] === args.oldContext['$implicit']) {
             args.view.detectChanges();
             const row = this.getRowByIndex(args.context.index);
-            if (row) {
+            if (row && row instanceof IgxRowComponent) {
```

Comment:
The reason for this issue is that we have a groupBy row and a grid row with the same index. We have to search only in the `dataRowList` like this:
`const row = this.dataRowList.find((r) => r.index === args.context.index);`

Reply:
Also I have noticed that there is no test added for this scenario. It would be nice if we have such.

Reply:
Well yes, this is the other way that could be fixed. I can change it.
There is already a test that covers this that for some was passing.

Reply:
It looks ok, but I was thinking if a similar issue comes up with the horizontal scrolling. It would be better to fix the row finding there as well.

Reply:
There is no issue for the horizontal scrolling because there is basically the same check I used earlier that covers these scenarios instead of parsing the dataRowList directly. Either way will work.